lib/dma.c

```diff
@@ -340,7 +367,23 @@ MOCK_DEFINE(dma_controller_add_region)(dma_controller_t *dma,
     region->fd = fd;
 
     if (fd != -1) {
-        int ret = dma_map_region(dma, region);
+        int ret;
+
+        /*
+         * TODO introduce a function that tells whether dirty page logging is
+         * enabled
+         */
+        if (dma->dirty_pgsize != 0) {
+            if (dirty_page_logging_start_on_region(region, dma->dirty_pgsize) < 0) {
+                /*
```

Comment:
I'm not sure it's worth fixing this just for a calloc() failure?

Reply:
I've noticed that dirty page tracking errors aren't fatal (e.g. worst that can happen is that migration fails but the VM keeps running on the source) but failing to add a DMA region would be. I wanted this documented to make sure we don't forget, shall I drop the `TODO` and keep the comment?